airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/jsonschema/AirbyteJsonSchemaType.kt

```diff
@@ -0,0 +1,215 @@
+                            NUMBER
+                        }
+                    }
+                    "string" -> {
+                        when (format) {
+                            "big_integer" -> INTEGER
+                            "float" -> NUMBER
+                            "date" -> DATE
+                            "time" -> {
+                                when (airbyteType) {
+                                    null,
+                                    "time_with_timezone" -> TIME_WITH_TIMEZONE
+                                    "time_without_timezone" -> TIME_WITHOUT_TIMEZONE
+                                    else ->
+                                        throw IllegalArgumentException(
+                                            "Unknown time format: $airbyteType"
+                                        )
+                                }
+                            }
+                            "date-time" -> {
+                                when (airbyteType) {
+                                    null,
+                                    "timestamp_with_timezone" -> TIMESTAMP_WITH_TIMEZONE
+                                    "timestamp_without_timezone" -> TIMESTAMP_WITHOUT_TIMEZONE
+                                    else ->
+                                        throw IllegalArgumentException(
+                                            "Unknown date-time format: $airbyteType"
+                                        )
+                                }
+                            }
+                            else -> STRING
+                        }
+                    }
+                    "array" -> {
+                        when {
+                            schema.has("items") -> {
+                                if (schema["items"].isArray) {
+                                    ARRAY_WITH_ITEMS
+                                } else {
+                                    ARRAY_WITH_ITEM
+                                }
+                            }
+                            else -> ARRAY_WITHOUT_ITEMS
+                        }
+                    }
+                    "object" -> {
+                        if (schema.has("properties")) {
+                            OBJECT_WITH_PROPERTIES
+                        } else {
+                            OBJECT_WITHOUT_PROPERTIES
+                        }
+                    }
+                    else -> throw IllegalArgumentException("Unknown schema type: $type")
+                }
+            } else if (schema.has("properties")) {
+                // Usually the root node
+                return OBJECT_WITH_PROPERTIES
+            } else {
+                throw IllegalArgumentException("Unspecified schema type")
+            }
+        }
+
+        fun getMatchingValueForType(value: JsonNode, options: ArrayNode): ObjectNode {
+            println("matching: $value against (arraynode) $options")
```

Comment:
don't forget to remove printlns before merging

Reply:
done